src/ui/public/persisted_log/persisted_log.js

```diff
@@ -7,9 +7,10 @@ define(function (require) {
     function PersistedLog(name, options) {
       options = options || {};
       this.name = name;
-      this.maxLength = options.maxLength || 0;
+      this.maxLength = options.maxLength;
```

Comment:
While invalid values here don't *break* anything, `_.take` seems to return 1 item if the second param is true. Should probably do some intelligent type casting here.

If you set it to `false` for invalid values, you can also just check for that. 

```js
this.maxLength = parseInt(options.maxLength);
if (this.maxLength === NaN) this.maxLength = false;
```

Then later check for `this.maxLength !== false` in the conditionals